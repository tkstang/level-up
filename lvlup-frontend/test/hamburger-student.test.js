import StudentSidenav from '../src/components/Student/nav/sidenav';
import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import lvlupApp from '../src/reducers/index';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

test('StudentSidenav should render', () => {
  const store = mockStore({ lvlupApp });
  const student = shallow(
    <StudentSidenav studentLoginInfo={{ id: 1 }} studentPointsAndCampus={{ currentTotal: 500, totalEarned: 500 }} store={store} />,
  );
  expect(shallowToJson(student)).toMatchSnapshot();
});
