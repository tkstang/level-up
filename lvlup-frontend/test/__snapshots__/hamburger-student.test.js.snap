exports[`test StudentSidenav should render 1`] = `
<Menu
  className="studentSidenav"
  inverted={true}
  vertical={true}>
  <MenuItem
    id="student-sidenav-info">
    <Image
      as="img"
      centered={true}
      shape="circular"
      size="tiny"
      ui={true} />
    <div
      className="userdiv">
      <h4 />
      <p>
        500  Points Earned
      </p>
      <p>
         
        500 Points Remaining
      </p>
      <p />
    </div>
  </MenuItem>
  <Link
    replace={false}
    to="/student/dashboard">
    <MenuItem>
      <Icon
        as="i"
        name="dashboard" />
      Dashboard
    </MenuItem>
  </Link>
  <Link
    replace={false}
    to="/student/challenges">
    <MenuItem>
      <Icon
        as="i"
        name="chevron up" />
      Challenges
    </MenuItem>
  </Link>
  <Link
    replace={false}
    to="/student/rewards">
    <MenuItem>
      <Icon
        as="i"
        name="gift" />
      Rewards
    </MenuItem>
  </Link>
</Menu>
`;
